src/JustSaying/Messaging/MessageSerialization/NewtonsoftSerializer.cs

```diff
@@ -5,6 +5,10 @@
 
 namespace JustSaying.Messaging.MessageSerialization;
 
+#if NET8_0_OR_GREATER
+[RequiresUnreferencedCode(Constants.SerializationUnreferencedCodeMessage)]
+[RequiresDynamicCode(Constants.SerializationDynamicCodeMessage)]
+#endif
 public class NewtonsoftSerializer : IMessageSerializer
```

Comment:
This is an assumption at the moment. Newtonsoft.Json isn't currently annotated for AOT/Trimming compatibility.